### ISSUE TITLE
i wana better result plz help

### ISSUE THREAD

**Issue by sanicle** [geekyutao/RN#30]
when i use my single gpu model (batchsize=8 epoch=4 th=0.8) it works like this: 
![1_0_output](https://user-images.githubusercontent.com/5432504/115048669-605c0780-9f0c-11eb-9510-16bfa313446c.png)
![1_0_gt](https://user-images.githubusercontent.com/5432504/115048700-69e56f80-9f0c-11eb-9ad5-77677650ee71.png)
![3_2_gt](https://user-images.githubusercontent.com/5432504/115048720-723daa80-9f0c-11eb-8951-e1184de21e88.png)
![3_2_output](https://user-images.githubusercontent.com/5432504/115048754-7d90d600-9f0c-11eb-8485-27fe52f1300c.png)

but when i use your pre-trained multigpu model (as the former issue method) it doesn't work:
![1_0_output](https://user-images.githubusercontent.com/5432504/115048983-c5176200-9f0c-11eb-8008-448b4d343f27.png)
![1_0_gt](https://user-images.githubusercontent.com/5432504/115048995-c9437f80-9f0c-11eb-9ee6-6c4822463eeb.png)

how can i get a better result just like what in your paper? 
or how can i make the pre-trained model work on my single gpu?

**Comment by sanicle** [comment]
**part of the training log :**
...
=> Epoch[4](226400/229995): Avg L1 loss: 0.166399 | G loss: 0.228385 | Avg D loss: 0.660457 || Timer: 0.7238 sec. | IO: 0.0021
=> Epoch[4](226500/229995): Avg L1 loss: 0.167660 | G loss: 0.229736 | Avg D loss: 0.659424 || Timer: 0.7246 sec. | IO: 0.0021
=> Epoch[4](226600/229995): Avg L1 loss: 0.165047 | G loss: 0.227178 | Avg D loss: 0.658189 || Timer: 0.7241 sec. | IO: 0.0024
=> Epoch[4](226700/229995): Avg L1 loss: 0.164854 | G loss: 0.226934 | Avg D loss: 0.661149 || Timer: 0.7243 sec. | IO: 0.0020
=> Epoch[4](226800/229995): Avg L1 loss: 0.163818 | G loss: 0.226001 | Avg D loss: 0.659541 || Timer: 0.7244 sec. | IO: 0.0021
=> Epoch[4](226900/229995): Avg L1 loss: 0.166666 | G loss: 0.229151 | Avg D loss: 0.659408 || Timer: 0.7245 sec. | IO: 0.0022
=> Epoch[4](227000/229995): Avg L1 loss: 0.161157 | G loss: 0.223243 | Avg D loss: 0.659248 || Timer: 0.7249 sec. | IO: 0.0022
=> Epoch[4](227100/229995): Avg L1 loss: 0.159167 | G loss: 0.221401 | Avg D loss: 0.660830 || Timer: 0.7242 sec. | IO: 0.0022
=> Epoch[4](227200/229995): Avg L1 loss: 0.168148 | G loss: 0.230427 | Avg D loss: 0.658346 || Timer: 0.7243 sec. | IO: 0.0025
=> Epoch[4](227300/229995): Avg L1 loss: 0.169166 | G loss: 0.231475 | Avg D loss: 0.658742 || Timer: 0.7237 sec. | IO: 0.0022
=> Epoch[4](227400/229995): Avg L1 loss: 0.162850 | G loss: 0.224701 | Avg D loss: 0.661460 || Timer: 0.7248 sec. | IO: 0.0021
=> Epoch[4](227500/229995): Avg L1 loss: 0.161034 | G loss: 0.223430 | Avg D loss: 0.658830 || Timer: 0.7231 sec. | IO: 0.0023
=> Epoch[4](227600/229995): Avg L1 loss: 0.160901 | G loss: 0.223161 | Avg D loss: 0.660453 || Timer: 0.7242 sec. | IO: 0.0025
=> Epoch[4](227700/229995): Avg L1 loss: 0.165468 | G loss: 0.227454 | Avg D loss: 0.659273 || Timer: 0.7233 sec. | IO: 0.0025
=> Epoch[4](227800/229995): Avg L1 loss: 0.163730 | G loss: 0.225986 | Avg D loss: 0.657400 || Timer: 0.7244 sec. | IO: 0.0024
=> Epoch[4](227900/229995): Avg L1 loss: 0.162113 | G loss: 0.224268 | Avg D loss: 0.659038 || Timer: 0.7242 sec. | IO: 0.0022
=> Epoch[4](228000/229995): Avg L1 loss: 0.167935 | G loss: 0.230006 | Avg D loss: 0.658061 || Timer: 0.8013 sec. | IO: 0.0027
=> Epoch[4](228100/229995): Avg L1 loss: 0.159820 | G loss: 0.222117 | Avg D loss: 0.660133 || Timer: 0.7467 sec. | IO: 0.0027



**test output:**
...
Total number of parameters: 2763776
Pre-trained G model is loaded.
===> Loading datasets
Total instance number: 3
===> Loaded datasets
Number: 00001  | Average: PSNR: 21.4464  SSIM: 0.8088  L1: 0.0244 | Current batch: 1  PSNR: 21.4464  SSIM: 0.8088  L1: 0.0244
Number: 00002  | Average: PSNR: 19.7406  SSIM: 0.6995  L1: 0.0420 | Current batch: 2  PSNR: 18.0348  SSIM: 0.5903  L1: 0.0595
Number: 00003  | Average: PSNR: 20.3026  SSIM: 0.7321  L1: 0.0370 | Current batch: 3  PSNR: 21.4264  SSIM: 0.7973  L1: 0.0271



**enviornment:**
Python 3.9.2
Pytorch 1.8.1
Cudatoolkit 11.1.1
newest scikit-image etc.

**Comment by geekyutao** [comment]
Did you see the ./render images during training? Is it normal? And have you tried using torch 1.1 ?

**Comment by sanicle** [comment]
render images is normal I can see all of them. better and better.. but the test result like what I pasted up there.

my torch is 1.8.1. because I use RTX3XXX that only support cuda11.x so I cannot use torch 1.1. What a pity T_T...
I rewrite some code to fix some small problems but the model structure is not changed and it works.

I'm still finding what the problem is.

**Comment by sanicle** [comment]
Even 6 epochs trained with single RTX3090. it still don't have the good result like what in your paper. There must be some other problem.

**Comment by geekyutao** [comment]
If the rendering is normal, then I believe there are some potential mistakes in evaluation. How much is the test PSNR now?

**Comment by sanicle** [comment]
PSNR is still less than 23. The training loss after that seems to have not declined. What is your final loss when PSNR up to 25?
and maybe I should set a smaller learning rate? I'm not quite sure.

**Comment by sanicle** [comment]
I stopped training just now because the loss is always the same it make no sense. I‘m setting another learning rate other than default and try again.

**Comment by geekyutao** [comment]
> I stopped training just now because the loss is always the same it make no sense. I‘m setting another learning rate other than default and try again.

The learning rate is proper now. I believe it is caused by other reasons.

**Comment by sanicle** [comment]
> > I stopped training just now because the loss is always the same it make no sense. I‘m setting another learning rate other than default and try again.
> 
> The learning rate is proper now. I believe it is caused by other reasons.

yes yr right. it is the same. PSNR is still less than 23. The training loss after epoch2-3 seems to have not declined.
I don't know how to solve it now.

**Comment by wsfwsf** [comment]
可能是eval.py中的
model.generator = torch.nn.DataParallel(model.generator, device_ids=gpus_list)
model.discriminator = torch.nn.DataParallel(model.discriminator, device_ids=gpus_list)
的问题，删除以后显示正常。
但是，我用celeba-HQ上实验，bs = 4,epoch = 30,结果还是比较模糊
